foreman/data_refinery_foreman/foreman/management/commands/create_compendia.py

```diff
@@ -65,33 +35,143 @@ def add_arguments(self, parser):
-        """
-        if options["organisms"] is None:
-            all_organisms = Organism.objects.exclude(name__in=["HOMO_SAPIENS", "MUS_MUSCULUS"])
-        else:
-            organisms = options["organisms"].upper().replace(" ", "_").split(",")
-            all_organisms = Organism.objects.filter(name__in=organisms)
-
-        # I think we could just use options["quant_sf_only"] but I
-        # wanna make sure that values that are not True do not trigger
-        # a truthy evaluation.
         quant_sf_only = options["quant_sf_only"] is True
 
         # default algorithm to arpack until we decide that ranomized is preferred
         svd_algorithm = 'NONE' if quant_sf_only else 'ARPACK'
         if options["svd_algorithm"] in ['ARPACK', 'RANDOMIZED', 'NONE']:
             svd_algorithm = options["svd_algorithm"]
 
-        logger.debug('Generating compendia for organisms', organisms=all_organisms)
+        logger.debug('Generating compendia for organisms', organism_groups=str(grouped_organisms))
 
-        for organism in all_organisms:
-            job = create_job_for_organism(organism, quant_sf_only, svd_algorithm)
-            logger.info("Sending CREATE_COMPENDIA for Organism", job_id=str(job.pk), organism=str(organism))
+        for organism_group in grouped_organisms:
+            job = create_job_for_organism(organism_group, quant_sf_only, svd_algorithm)
+            logger.info("Sending CREATE_COMPENDIA for Organism", job_id=str(job.pk),
+                        organism_group=str(organism_group))
             send_job(ProcessorPipeline.CREATE_COMPENDIA, job)
 
         sys.exit(0)
+
+    def _get_target_organisms(self, options):
+        all_organisms = list(get_compendia_organisms().values_list('id', flat=True))
+        all_organisms = Organism.objects.filter(id__in=all_organisms)
+
+        if options["organisms"] is None:
+            target_organisms = all_organisms.exclude(name__in=["HOMO_SAPIENS", "MUS_MUSCULUS"])
+        else:
+            organisms = options["organisms"].upper().replace(" ", "_").split(",")
+            target_organisms = all_organisms.filter(name__in=organisms)
+
+        return target_organisms
+
+
+def get_compendia_organisms():
+    """ We start with the organisms that have QN targets associated with them.
+    Here we are filtering out the ones with invalid QN targets. This can be removed once
+    https://github.com/AlexsLemonade/refinebio/issues/1757 is addressed """
+    # We have some invalid qn targets, that's why we check that each organism needs to have
+    # at least 100 processed samples. This condition can be removed once
+    # https://github.com/AlexsLemonade/refinebio/issues/1757 is addressed
+    return Organism.objects\
+        .filter(qn_target__isnull=False)\
+        .annotate(num_samples=Count('sample', filter=Q(sample__is_processed=True, sample__technology='MICROARRAY'), distinct=True))\
+        .filter(num_samples__gt=100)
```

Comment:
This isn't quite going to do it. It's not just 100 microarray samples, it's 100 of them on the same platform.

Reply:
That check will be redundant once the invalid QN targets are removed in https://github.com/AlexsLemonade/refinebio/issues/1757. Note we are only using the organisms that have valid QN Targets, and for those, we know that have >100 samples on the same platform.

Do you think we should make this more explicit here? Or maybe wait until the invalid QN targets are removed before merging this?

Reply:
Well I don't know about waiting before merging this. As it is now we'll already create compendia jobs for organisms that shouldn't have it so we aren't getting worse.

However I think we should either remove this check because it won't be necessary after #1757 or we should do it right. It doesn't look like we prioritized #1757 for this sprint though, so unless we want to do that it looks like it could be an issue for a bit?

Reply:
What if we make a note on #1757 to remove the additional check once that is addressed? As it is now, this will generate the groups at https://github.com/AlexsLemonade/refinebio/issues/1736#issuecomment-541183861 which all seem like will be valid compendiums.

Reply:
Oh wait, https://github.com/AlexsLemonade/refinebio/pull/1759 is already open. Let's just get that out and this'll be a non-issue. I'm reviewing it now.